CMakeLists.txt

```diff
@@ -91,6 +74,56 @@ if(ENABLE_MKLCPU_BACKEND
   list(APPEND DOMAINS_LIST "rng")
 endif()
 
+# Define required CXX compilers before project
+if(CMAKE_CXX_COMPILER)
+  if(WIN32)
+    string(REPLACE "\\" "/" CMAKE_CXX_COMPILER ${CMAKE_CXX_COMPILER})
+  endif()
+else()
+  if(ENABLE_CUBLAS_BACKEND OR ENABLE_CURAND_BACKEND)
+    set(CMAKE_CXX_COMPILER "clang++")
+  elseif(ENABLE_MKLGPU_BACKEND)
+    set(CMAKE_CXX_COMPILER "dpcpp")
+  else()
+    find_program(DPCPP_PATH dpcpp)
+    if(DPCPP_PATH)
+      set(CMAKE_CXX_COMPILER "dpcpp")
```

Comment:
Can you please share the purpose of using `find_program` here?

Reply:
I use `find_program` to check if `dpcpp` is available, so `dpcpp` has a higher precedence than open source compilers in this implementation.

Reply:
Does CMake not find `dpcpp` from only line 90? CMake should do that search automatically and find `dpcpp` in the environment variables like `PATH`. Can you check this code section without find_program and if-check?

Reply:
Line 90 is not sufficient. The `set` command only sets the variable literally. The search and check of compilers are triggered by `project`.

Reply:
okay thanks!